packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -206,4 +213,62 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
   bool _isImageUploadedForType(ImageField imageType) {
     return (_uploadedImages[imageType] ?? <File>[]).isNotEmpty;
   }
+
+  Widget _buildNutritionInputButton() {
+    if (_nutritionFactsAdded) {
+      return Padding(
+          padding: _ROW_PADDING_TOP,
+          child: Row(
+            mainAxisSize: MainAxisSize.max,
+            mainAxisAlignment: MainAxisAlignment.start,
+            children: <Widget>[
+              const SizedBox(
+                width: 50.0,
+                child: Icon(
+                  Icons.check,
+                  color: Colors.greenAccent,
+                ),
+              ),
+              Expanded(
+                child: Center(
+                  child: Text(
+                      AppLocalizations.of(context)!.nutritional_facts_added,
+                      style: Theme.of(context).textTheme.bodyText1),
+                ),
+              ),
+            ],
+          ));
+    }
+
+    return Padding(
+      padding: _ROW_PADDING_TOP,
+      child: SmoothLargeButtonWithIcon(
+        text:
+            AppLocalizations.of(context)!.nutritional_facts_input_button_label,
+        icon: Icons.edit,
+        onPressed: () async {
+          final OrderedNutrientsCache? cache =
+              await OrderedNutrientsCache.getCache(context);
+          if (cache == null) {
+            return;
+          }
```

Comment:
Is this a somewhat likely case, if so a Snackbar or a todo comment would be good